wandb/integration/openai/fine_tune.py

```diff
@@ -0,0 +1,381 @@
+
+        :param fine_tune_job_id: The id of the fine-tune (optional)
+        :param openai_client: Pass the `OpenAI()` client (optional)
+        :param first_n_fine_tunes: Number of most recent fine-tunes to log when an fine_tune_job_id is not provided. By default, every fine-tune is synced.
+        :param project: Name of the project where you're sending runs. By default, it is "GPT-3".
+        :param entity: Username or team name where you're sending runs. By default, your default entity is used, which is usually your username.
+        :param overwrite: Forces logging and overwrite existing wandb run of the same fine-tune.
+        :param wait_for_job_success: Waits for the fine-tune to be complete and then log metrics to W&B. By default, it is True.
+        """
+        if openai_client is None:
+            openai_client = OpenAI(
+                api_key=os.environ["OPENAI_API_KEY"],
+            )
+        cls.openai_client = openai_client
+
+        if fine_tune_job_id:
+            wandb.termlog("Retrieving fine-tune job...")
+            fine_tune = openai_client.fine_tuning.jobs.retrieve(
+                fine_tuning_job_id=fine_tune_job_id
+            )
+            fine_tunes = [fine_tune]
+        else:
+            # get list of fine_tune to log
+            fine_tunes = openai_client.fine_tuning.jobs.list()
+            if not fine_tunes or fine_tunes.data is None:
+                wandb.termwarn("No fine-tune has been retrieved")
+                return
+            # Select the `first_n_fine_tunes` from the `fine_tunes.data` list.
+            # If `first_n_fine_tunes` is None, it selects all items in the list (from start to end).
+            # If for example, `first_n_fine_tunes` is 5, it selects the last 5 items in the list.
+            # Note that the last items in the list are the latest fine-tune jobs.
+            fine_tunes = fine_tunes.data[
+                -first_n_fine_tunes if first_n_fine_tunes is not None else None :
+            ]
+
+        # log starting from oldest fine_tune
+        show_individual_warnings = not (
+            fine_tune_job_id is None and first_n_fine_tunes is None
+        )
+        fine_tune_logged = []
+        for fine_tune in fine_tunes:
+            if wait_for_job_success:
+                fine_tune = cls._wait_for_job_success(fine_tune)
+
+            cls._log_fine_tune(
+                fine_tune,
+                project,
+                entity,
+                overwrite,
+                show_individual_warnings,
+                **kwargs_wandb_init,
+            )
+
+        if not show_individual_warnings and not any(fine_tune_logged):
+            wandb.termwarn("No new successful fine-tunes were found")
+
+        return "🎉 wandb sync completed successfully"
+
+    @classmethod
+    def _wait_for_job_success(cls, fine_tune: FineTuningJob) -> FineTuningJob:
+        wandb.termwarn("Waiting for the Fine-tuning job to be finished...")
```

Comment:
this one doesn't seem like a warning to me... seem more like info, so probably use `wandb.termlog`

Reply:
thanks for catching this. fixing.